File: data/src/main/java/org/cryptomator/data/cloud/onedrive/OnedriveFile.java
package org.cryptomator.data.cloud.onedrive;

import org.cryptomator.domain.Cloud;
import org.cryptomator.domain.CloudFile;
import org.cryptomator.util.Optional;

import java.util.Date;

class OnedriveFile implements CloudFile, OnedriveNode {

	private final OnedriveFolder parent;
	private final String name;
	private final String path;
	private final Optional<Long> size;
	private final Optional<Date> modified;

	public OnedriveFile(OnedriveFolder parent, String name, String path, Optional<Long> size, Optional<Date> modified) {
		this.parent = parent;
		this.name = name;
		this.path = path;
		this.size = size;
		this.modified = modified;
	}

	@Override
	public boolean isFolder() {
		return false;
	}

	@Override
	public Cloud getCloud() {
		return parent.getCloud();
	}

	@Override
	public String getName() {
		return name;
	}

	@Override
	public String getPath() {
		return path;
	}

	@Override
	public OnedriveFolder getParent() {
		return parent;
	}

	@Override
	public Optional<Long> getSize() {
		return size;
	}

	@Override
	public Optional<Date> getModified() {
		return modified;
	}
}

File: data/src/apiKey/java/org/cryptomator/data/cloud/onedrive/OnedriveFile.kt
package org.cryptomator.data.cloud.onedrive

import org.cryptomator.domain.Cloud
import org.cryptomator.domain.CloudFile
import java.util.Date

internal class OnedriveFile(override val parent: OnedriveFolder, override val name: String, override val path: String, override val size: Long?, override val modified: Date?) : CloudFile, OnedriveNode {

	override val isFolder: Boolean = false

	override val cloud: Cloud?
		get() = parent.cloud
}
